bench/src/benchmark_result.rs

```diff
@@ -49,10 +90,43 @@ impl Display for BenchmarkResults {
         let total_throughput = total_size_bytes as f64 / total_duration / 1024.0 / 1024.0;
         let messages_per_second = total_messages as f64 / total_duration;
 
-        write!(
+        BenchStatistics {
+            total_throughput,
+            messages_per_second,
+            average_latency,
+            average_throughput,
+            total_duration,
+        }
+    }
+}
+
+impl Display for BenchmarkResults {
+    fn fmt(&self, f: &mut Formatter<'_>) -> std::fmt::Result {
+        if let Some(test_type) = self.get_test_type() {
+            if test_type == BenchmarkKind::SendAndPoll {
+                let producer_statics = self.calculate_statitics(|x| x.kind == BenchmarkKind::Send);
+                let consumer_statics = self.calculate_statitics(|x| x.kind == BenchmarkKind::Poll);
+
+                writeln!(
+                        f,
+                        "\x1B[32mProducer results: total throughput: {:.2} MB/s, {:.0} messages/s, average latency: {:.2} ms, average throughput: {:.2} MB/s, total duration: {:.2} s\x1B[0m",
```

Comment:
Check if https://crates.io/crates/colour or anything similar could be used for formatting / colours instead of manual setting

Reply:
Done